### PR TITLE
Delete DID wallet after transfer

### LINKED ISSUE #13705: [Bug] DID metadata updates not included in spendbundle transmitted to the blockchain
### What happened?

While DID metadata updates are updated in local wallet storage, the innerpuz put into the spendbundle by create_update_spend() does not include it

in https://github.com/Chia-Network/chia-blockchain/blob/main/chia/wallet/did_wallet/did_wallet.py#L1318

this would be a good test to be added to CI

### Version

1.6.0

### What platform are you using?

Linux

### What ui mode are you using?

CLI

### Relevant log output

```shell
To replicate:

create DID (w/ fee = 5 mojos+)

update metadata of DID  (w/ fee = 5 mojos+)

get metadata locally

get DID coin info

get metadata from the blockchain using full node api 

then parse the DID metadata from the blockchain
```

### PR DESCRIPTION
**Pull request by ytx1991**

### CODE REVIEW

File: chia/wallet/wallet_state_manager.py [hunk 1]
```diff
@@ -775,13 +775,25 @@ async def handle_did(
...
+        launch_id: bytes32 = bytes32(bytes(singleton_struct.rest().first())[1:])
         if derivation_record is None:
             self.log.info(f"Received state for the coin that doesn't belong to us {coin_state}")
+            # Check if it was owned by us
+            removed_wallet_ids = []
+            for wallet in self.wallets.values():
+                if (
+                    wallet.type() == WalletType.DECENTRALIZED_ID
+                    and launch_id == wallet.did_info.origin_coin.name()
+                    and not wallet.did_info.sent_recovery_transaction
```
Comment (trepca) [inline chia/wallet/wallet_state_manager.py:787]: what does `sent_recovery_transaction` represent?

Comment (ytx1991) [reply chia/wallet/wallet_state_manager.py:787]: If it is created by recovery

**Review by trepca — DISMISSED** [review]
please remove `rl_wallet` files

**Comment by ytx1991** [comment]
> please remove `rl_wallet` files

The pre-commit will fail

**Comment by trepca** [comment]
can we add a test for DID recovery hints?

**Review by trepca — DISMISSED** [review]

File: chia/wallet/wallet_state_manager.py [hunk 2]
```diff
@@ -774,13 +775,26 @@ async def handle_did(
...
+                did_info: DIDInfo = DIDInfo.from_json_dict(json.loads(wallet_info.data))
+                if (
+                    did_info.origin_coin is not None
+                    and launch_id == did_info.origin_coin.name()
+                    and not did_info.sent_recovery_transaction
+                ):
+                    await self.user_store.delete_wallet(wallet_info.id)
+                    removed_wallet_ids.append(wallet_info.id)
+            for remove_id in removed_wallet_ids:
+                self.wallets.pop(remove_id)
```
Comment (aqk) [inline chia/wallet/wallet_state_manager.py:793]: This should probably also do:

```suggestion
                self.wallets.pop(remove_id)
                await self.user_store.delete_wallet(remove_id)
```

**Review by aqk — DISMISSED** [review]
Take a peek at the `self.wallets.pop(remove_id)` line and let me know what you think.

File: chia/wallet/wallet_state_manager.py (hunk 2 above)
Comment (ytx1991) [reply chia/wallet/wallet_state_manager.py:793]: It is deleted at ln790

**Comment by ytx1991** [comment]
An urgent fix needs to be added, hold on merge

**Review by trepca — DISMISSED** [review]

**Review by trepca — DISMISSED** [review]

**Review by trepca — APPROVED** [review]

**Pull request merged**